tests/testthat/test_get_stan_data.R

```diff
@@ -0,0 +1,214 @@
+forecast_horizon <- 28
+include_ww <- 1
+
+input_count_data <- get_input_count_data_for_stan(
+  count_data,
+  calibration_time
+)
+first_count_data_date <- min(input_count_data$date, na.rm = TRUE)
+last_count_data_date <- max(input_count_data$date, na.rm = TRUE)
+input_ww_data <- get_input_ww_data_for_stan(
+  ww_data_filtered,
+  first_count_data_date,
+  last_count_data_date,
+  calibration_time
+)
+
+
+test_that(paste0(
+  "Test that modifying calibration time generates data of expected",
+  " length"
+), {
+  result <- get_input_count_data_for_stan(
+    count_data,
+    calibration_time = 80
+  )
+  expect_true(nrow(result) == 80)
+})
+
+test_that(paste0(
+  "Test that passing out of window wastewater data behaves as",
+  "expected"
+), {
+  # Make wastewater data outside of scope of admissions data
+  recent_ww_data <- tibble::tibble(
+    date = rep(seq(
+      from = lubridate::ymd("2024-08-01"),
+      to = lubridate::ymd("2024-11-01"),
+      by = "weeks"
+    ), 2),
+    site = c(rep(1, 14), rep(2, 14)),
+    lab = c(rep(1, 28)),
+    conc = abs(rnorm(28, mean = 500, sd = 50)),
+    lod = c(rep(20, 14), rep(15, 14)),
+    site_pop = c(rep(2e5, 14), rep(4e5, 14))
+  )
+
+  recent_ww_data_preprocessed <- preprocess_ww_data(recent_ww_data,
+    conc_col_name = "conc",
+    lod_col_name = "lod"
+  )
+  recent_input_ww_data <- indicate_ww_exclusions(recent_ww_data_preprocessed)
+
+  recent_input_ww_data_for_stan <- get_input_ww_data_for_stan(
+    recent_input_ww_data,
+    first_count_data_date,
+    last_count_data_date,
+    calibration_time
+  )
+
+  expect_error(get_stan_data(
+    input_count_data,
+    recent_input_ww_data_for_stan,
+    forecast_date,
+    forecast_horizon,
+    calibration_time,
+    generation_interval,
+    inf_to_count_delay,
+    infection_feedback_pmf,
+    params,
+    include_ww
+  ))
+
+  # Make wastewater data outside of scope of admissions data
+  old_ww_data <- tibble::tibble(
+    date = rep(seq(
+      from = lubridate::ymd("2022-08-01"),
+      to = lubridate::ymd("2022-11-01"),
+      by = "weeks"
+    ), 2),
+    site = c(rep(1, 14), rep(2, 14)),
+    lab = c(rep(1, 28)),
+    conc = abs(rnorm(28, mean = 500, sd = 50)),
+    lod = c(rep(20, 14), rep(15, 14)),
+    site_pop = c(rep(2e5, 14), rep(4e5, 14))
+  )
+
+  old_ww_data_preprocessed <- preprocess_ww_data(old_ww_data,
+    conc_col_name = "conc",
+    lod_col_name = "lod"
+  )
+  old_input_ww_data <- indicate_ww_exclusions(old_ww_data_preprocessed)
+
+  old_input_ww_data_for_stan <- get_input_ww_data_for_stan(
+    old_input_ww_data,
+    first_count_data_date,
+    last_count_data_date,
+    calibration_time
+  )
+
+
+  expect_error(get_stan_data(
+    input_count_data,
+    old_input_ww_data,
+    forecast_date,
+    forecast_horizon,
+    calibration_time,
+    generation_interval,
+    inf_to_count_delay,
+    infection_feedback_pmf,
+    params,
+    include_ww
+  ))
+})
+
+test_that("Test that pmf check works as expected", {
```

Comment:
Would be good to have a unit test for the PMF validation function itself in`test_checkers.R`. But checking that it gets called here and behaves as expected in context is nice to have too.